src/mds/MDSMap.cc

```diff
@@ -780,26 +731,21 @@ void MDSMap::decode(bufferlist::const_iterator& p)
   __u16 ev = 1;
   if (struct_v >= 2)
     decode(ev, p);
+  if (decode_ev)
+    ceph_assert(ev == 5);
   if (ev >= 3)
     decode(compat, p);
   else
     compat = get_compat_set_base();
-  if (ev < 5) {
-    __u32 n;
-    decode(n, p);
-    metadata_pool = n;
-  } else {
-    decode(metadata_pool, p);
-  }
+
   decode(created, p);
   decode(modified, p);
-  decode(tableserver, p);
```

Comment:
can't remove this

Reply:
Fixed this. What about `metadata_pool` then?